app/server/lib/ConfigBackendAPI.ts

```diff
@@ -0,0 +1,67 @@
+import * as express from 'express';
+import {expressWrap} from 'app/server/lib/expressWrap';
+
+import {IGristCoreConfig} from 'app/server/lib/configCore';
+import {getGlobalConfig} from 'app/server/lib/globalConfig';
+
+import log from "app/server/lib/log";
+
+export class ConfigBackendAPI {
+  private static _doRestart = false;
+  private _config: IGristCoreConfig;
+
+  constructor() {
+      this._config = getGlobalConfig();
+  }
+
+  public addEndpoints(app: express.Express, requireMiddleware: express.RequestHandler) {
+    app.get('/api/config/:key', requireMiddleware, expressWrap((req, resp) => {
+      log.debug(`Scheduled to do a config restart is ${ConfigBackendAPI._doRestart}`);
+      if (ConfigBackendAPI._doRestart) {
+        resp.on('finish', () => {
+          // If we have IPC with parent process (e.g. when running under
+          // Docker) tell the parent that it's time to restart with a
+          // new config.
+          if (process.send) {
+            log.debug('config: requesting supervisor to restart the Grist server');
+            process.send({ action: 'restart' });
+          }
+        });
+        // We might not have restarted (maybe we aren't running with a
+        // parent process), but either way, don't try to restart
+        // again.
+        ConfigBackendAPI._doRestart = false;
+      }
+      log.debug('config: requesting configuration', req.params);
+
+      // Only one key is valid for now
+      if (req.params.key === 'edition') {
+        resp.send({value: this._config.edition.get()});
+      } else {
+        resp.status(404).send('Configuration key not found.');
+      }
+    }));
+
+    app.patch('/api/config', requireMiddleware, expressWrap(async (req, resp) => {
+
+      const config = req.body.config;
+      log.debug('config: received new configuration item', config);
+
+      if (req.body.restart) {
+        // We do the restart in the GET above, because the frontend
+        // wants to immediately do a GET after a PATCH, so we still
+        // need to provide an answer before we restart the server.
+        ConfigBackendAPI._doRestart = true;
+      }
+
+      // Only one key is valid for now
+      if(config.edition !== undefined) {
+        await this._config.edition.set(config.edition);
+
+        resp.sendStatus(200);
```

Comment:
I don't like this. This endpoint accepts json, and should return json, not a text.